### ISSUE TITLE
MAL Heatmap no longer getting updates since beginning of January

### ISSUE THREAD

**Issue by TheGreatDrRodneyMcKay** [andersonkrs/malheatmap#617]
Hello, 
it seems like MAL heatmap hasn't been updating lists ever since the beginning of January (around the 3rd or something).
You can use my list as an example but do keep in in mind I've tried on several other accounts with public access (tried in guest mode so I'm sure of it).
My MAL: https://myanimelist.net/history/DrRodneyMcKay
My MALHeatMap: https://malheatmap.com/users/DrRodneyMcKay

**Comment by SuryanshChopra** [comment]
Hi @TheGreatDrRodneyMcKay try reach them on their support mail [hello@malheatmap.com](mailto:hello@malheatmap.com).

**Comment by KanchiMoe** [comment]
Can also confirm this issue.

**Comment by andersonkrs** [comment]
Sorry folks, MAL blocked us, not sure when they will unblock, I need some time to move out the server to another IP, in the mean time I deactivated the service of syncing

**Comment by KanchiMoe** [comment]
@andersonkrs it seems it has started working again!

**Comment by SuryanshChopra** [comment]
Thanks for the update, can confirm the same. Thanks @andersonkrs for doing this !

**Comment by andersonkrs** [comment]
Hey all, first of all my apologies for the inconvenience.

I've moved the service to another machine, but after couple ours they blocked us again, I just deployed a trick to fake being a different device each time, but I don't know how long it will last this workaround.

So the sync service can stop at any time again,

I have plans to migrate over consuming the MAL's official API, in general it will be better because the chance of being blocked is very little if I respect the rate limiting of the API. Also it will make things easier, when users change their data (like user name) we will be able to sync it back to heatmap without losing the link.

This also brings challenges, like they don't expose the user history in their API, so I'll need to generate the activities based on a snapshot of the [anime/manga list](https://myanimelist.net/apiconfig/references/api/v2#operation/users_user_id_mangalist_get) that I'll be capturing periodically throughout the day

I don't have an ETA, just wanna let you all know that good stuff is coming.
